tree/treeplayer/inc/ROOT/TDFUtils.hxx

```diff
@@ -35,6 +35,16 @@ namespace ROOT {
 namespace Experimental {
 template <int D, typename P, template <int, typename, template <typename> class> class... S>
 class THist;
+
+namespace TDF {
+struct SnapshotOptions {
+  Int_t fAutoFlush  = 0;          //< AutoFlush value for output tree
+  Int_t fCompress   = 1;          //< Compression algorithm and compression level of output file
```

Comment:
Don't repeat the mistakes of the past!  It is nasty and not-type-safe to combine multiple pieces of information into a single integer.
- Compression algorithm should be from an enum class.
- Compression level should be a separate integer and enforce the limits.

Reply:
In principle I agree with you, but I'd like to hear @pcanal's opinion before moving on with these changes.